android/titanium/src/java/org/appcelerator/titanium/proxy/ColorProxy.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Appcelerator Titanium Mobile
+ * Copyright (c) 2020-Present by Axway, Inc. All Rights Reserved.
+ * Licensed under the terms of the Apache Public License
+ * Please see the LICENSE included with this distribution for details.
+ */
+package org.appcelerator.titanium.proxy;
+
+import org.appcelerator.kroll.KrollProxy;
+import org.appcelerator.kroll.annotations.Kroll;
+
+import android.graphics.Color;
+
+import androidx.annotation.ColorInt;
+
+@Kroll.proxy
+/**
+ * This is a proxy representation of the Android Color type.
+ * Refer to <a href="https://developer.android.com/reference/android/graphics/Color">Android Color</a> for more details.
+ */
+public class ColorProxy extends KrollProxy
+{
+	private Color color;
+
+	public ColorProxy(@ColorInt int colorInt)
+	{
+		this(Color.valueOf(colorInt));
+	}
+
+	public ColorProxy(Color color)
+	{
+		this.color = color;
+	}
+
+	@Kroll.method
+	public String toHex()
+	{
+		if (this.color == null) {
+			return "#000000";
+		}
+		// Padding! If length is less than 8, pad with leading 0s
+		String hex = Integer.toHexString(this.color.toArgb());
+		while (hex.length() < 8) {
+			hex = "0" + hex;
+		}
+		return "#" + hex;
+	}
```

Comment:
I think we should return RGBA like how iOS used to do it.

```java
// Get ARGB color value.
int intValue = this.color.toArgb();

// Convert to RGBA hex string.
String hexString = String.format(
	"#%02X%02X%02X%02X",
	(intValue >>> 16) & 0xFF,
	(intValue >>> 8) & 0xFF,
	intValue & 0xFF,
	intValue >>> 24);
return hexString;
```

Reply:
Oh hold on. We can't use the `Color.toArgb()` method. It's API Level 26. It will crash on OS versions older than Android 8.0.
https://developer.android.com/reference/android/graphics/Color#toArgb()

So that leaves the `red()`, `green()`, `blue()`, and `alpha()` methods which unfortunately returns the components values as normalized `0.0-1.0` float values. Hmm... using a `Color` object member variable might not be worth it. Pretty much all Android APIs take a raw color `int` value anyways.